Playbooks/AS-AI-Commandline-Analysis/azuredeploy.json

```diff
@@ -0,0 +1,288 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "metadata": {
+        "Comments":  "This playbook is intended to be run from a Microsoft Sentinel incident that has a Commandline entity. It will send a prompt to GPT-3 (text-davinci-003 model) and return an explanation of the commandline to the comments of the incident.",
+        "Author": "Curtis Middlehurst"
+    },
+    "parameters": {
+        "playbook_name": {
+            "defaultValue": "",
+            "type": "string"
+        },
+        "userprincipalname":{
+            "defaultValue": "user@domain",
+            "type": "string"
+        }
+    },
+    "variables": {
+        "AzureSentinelConnectionName": "[concat('azuresentinel-', parameters('playbook_name'))]",
+        "OpenaiConnectionName": "[concat('openaiip-', parameters('playbook_name'))]"
+    },
+    "resources": [
+        {
+            "type": "Microsoft.Web/connections",
+            "apiVersion": "2016-06-01",
+            "name": "[variables('AzureSentinelConnectionName')]",
+            "location": "[resourceGroup().location]",
+            "properties": {
+                "displayName": "[parameters('userprincipalname')",
+                "customParameterValues": {},
+                "api": {
+                    "id": "[concat('/subscriptions/', subscription().subscriptionId, '/providers/Microsoft.Web/locations/', resourceGroup().location,'/managedApis/azuresentinel')]"
+                }
```

Comment:
I would suggest using Managed Identity for Sentinel connection. It will remove the requirement for user principal name and experience is more user friendly.  Below screenshot suggests what you need to add to enabled MI based authentication. Taken from this playbook, https://github.com/Azure/Azure-Sentinel/blob/master/Solutions/Neustar%20IP%20GeoPoint/Playbooks/Neustar-GetIPGeoInfo/azuredeploy.json
![delete](https://user-images.githubusercontent.com/8243910/216214740-08f63569-19da-43a9-8007-5f5d50a261ec.jpg)

Reply:
Thanks for the feedback on this. As I am working on the changes I have also observed the following has to be added to make the managed identity work by setting the identity of the playbook to system assigned.
![identityfield](https://user-images.githubusercontent.com/45267272/216281410-817648d0-9d9d-4195-9466-6ec2536db298.png)

Reply:
Resolved in 40730a1